File: CPP/7ReverseInteger.cpp
#include <iostream>
#include <cmath>

using namespace std;

class Solution {
public:
    int reverse(int x) {
    	long long ans=0;
    	const int max = 0x7fffffff;    
        const int min = 0x80000000;   
		while(x!=0){
			ans = ans*10 + x%10;
			if(ans > max || ans < min){
				return 0;
			}
			x /= 10;
		}
		return ans;    
    }
};

int main(){
	Solution s;
	cout << s.reverse(123) << endl;
	cout << s.reverse(-12) << endl;
	cout << s.reverse(100) << endl;
	cout << s.reverse(1534236469) << endl;
	cout << 0x7fffffff << " " <<  0x80000000 << endl; 
	return 0;
}

File: Python/7ReverseInteger.py
__author__ = 'Minyi'
class Solution(object):
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        postive = True
        if x < 0:
            postive = False
        x = abs(x)
        ans = 0
        max = 0x7fffffff
        min = 0x80000000
        while x!=0:
            ans = ans*10 + x%10
            if ans > 2 ** 31 - 1:
                return 0
            x //= 10
        if not postive:
            return -1*ans
        return ans